### PR TITLE
Compute: parallelize execution

### PR DESCRIPTION
**Pull request by camdencheek**
This parallelizes execution of compute jobs using the new `group.OrderedParallel`,
which was introduced in #35572. 

This dramatically speeds up execution time for compute queries that require fetching
the file contents. See [comparison video](https://vimeo.com/710511004). 

## Test plan

Running backend integration tests. Also manually tested a few queries to validate that the results are the consistent. 

Edit to bump pr-auditor

### CODE REVIEW

File: enterprise/cmd/frontend/internal/compute/streaming/compute.go [hunk 1]
```diff
@@ -11,27 +11,26 @@ import (
...
-			for _, diffMatch := range v.CommitToDiffMatches() {
-				result, err := cmd.Run(ctx, db, diffMatch)
-				if err != nil {
-					return err
-				}
-				f(result)
-			}
-		} else {
-			result, err := cmd.Run(ctx, db, m)
+func toComputeResult(ctx context.Context, db database.DB, cmd compute.Command, match result.Match) (out []compute.Result, _ error) {
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/streaming/compute.go:17]: minor change here: this only takes one result and does not stream them back, just returns them. Streaming is now handled by the parallel group

File: enterprise/cmd/frontend/internal/compute/streaming/compute.go [hunk 2]
```diff
@@ -45,20 +44,29 @@ func NewComputeStream(ctx context.Context, db database.DB, query string) (<-chan
 		return nil, func() (*search.Alert, error) { return nil, err }
 	}
 
-	eventsC := make(chan Event)
+	eventsC := make(chan Event, 8)
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/streaming/compute.go:47]: unrelated, but this really feels like it should be buffered, even minimally

File: enterprise/cmd/frontend/internal/compute/streaming/compute.go [hunk 3]
```diff
@@ -45,20 +44,29 @@ func NewComputeStream(ctx context.Context, db database.DB, query string) (<-chan
...
-			err = toComputeResultStream(ctx, db, computeQuery.Command, event.Results, callback)
-			if err != nil {
-				select {
-				case errorC <- err:
-				default:
-				}
+	type groupEvent struct {
+		results []compute.Result
+		err     error
+	}
```
Comment (camdencheek) [inline enterprise/cmd/frontend/internal/compute/streaming/compute.go:52]: this is just getting around the fact that go doesn't have tuple types 🙃

**Review by rvantonder — APPROVED** [review]
exciting.

**Pull request merged**